### PR TITLE
fix(component-testing,vite-dev-server): ensure to call unmount after each test

### PR DESCRIPTION
**Pull request by lmiller1990**
- Closes https://cypress-io.atlassian.net/browse/CT-318

### User facing changelog

N/A

### Additional details

We need to unmount the previous node when using Vite or strange things happen. We do this in `beforeEach`.

### How has the user experience changed?

It works properly now.

### PR Tasks

N/A

### CODE REVIEW

File: npm/vite-dev-server/index-template.html [hunk 1]
```diff
@@ -38,7 +38,7 @@
       Cypress.onSpecWindow(window, importsToLoad)
       Cypress.action('app:window:before:load', window)
 
-      beforeEach(() => {
+      before(() => {
```
Comment (lmiller1990) [inline npm/vite-dev-server/index-template.html:41]: This was causing problems. `beforeEach` breaks things. :shrug:

File: npm/vite-dev-server/cypress/support/support.js [hunk 2]
```diff
@@ -1,3 +1,9 @@
+import { unmount } from '@cypress/react'
+
 before(() => {
   window.supportFileWasLoaded = true
 })
+
+beforeEach(() => {
```
Comment (dmtrKovalenko) [inline npm/vite-dev-server/cypress/support/support.js:7]: Not sure that it should live in vite dev server. Probably let's move it to to `@cypress/react` itself

**Review by dmtrKovalenko — CHANGES_REQUESTED** [review]
Please before merging it can you please create a mostly real-world example in `react/examples` just how the real user will create it. In order to be sure that it will work for the standalone project and will correctly run on CI

File: npm/vite-dev-server/cypress/support/support.js (hunk 2 above)
Comment (lmiller1990) [reply npm/vite-dev-server/cypress/support/support.js:7]: I will try it

Comment (lmiller1990) [reply npm/vite-dev-server/cypress/support/support.js]: Ok, done, it works

Comment (lmiller1990) [reply npm/vite-dev-server/cypress/support/support.js]: Great, works fine

**Comment by elevatebart** [comment]
Can I have steps to test this? I have more failing tests in this branch when running the following command than I have in `develop`:

```sh
yarn workspace @cypress/vite-dev-server cy:run
```

**Review by elevatebart — APPROVED** [review]
LGTM

**Comment by elevatebart** [comment]
@lmiller1990 you might have to create a new PR for this one because @dmtrKovalenko will be blocking it until he comes back in 2 weeks.

**Comment by lmiller1990** [comment]
Closed in favor of https://github.com/cypress-io/cypress/pull/15385

**Pull request closed without merging**